Parameterized makefile

BenderScript opened the pull request:
- Preparing makefile for new project layout
- Included parameters for mostly everything so it is easy to change to new project layout
- Dockerfile creation/removal is under Makefile now
- No need for tools/test/run.sh anymore

BenderScript committed:
Parameterized makefile

- Preparing makefile for new project layout
- Included parameters for mostly everything so it is easy to change to new project layout
- Dockerfile creation/removal is under Makefile now
- No need for tools/test/run.sh anymore

remingtonc commented on Makefile:
```diff
@@ -1,8 +1,30 @@
+
+clean-containers:
+	@echo "  >  Cleaning containers"
+	@cd $(DOCKER) && docker-compose kill && docker-compose rm -f
```
Can simplify to `docker-compose down` ([doc](https://docs.docker.com/compose/reference/down/)).

BenderScript replied:
Done

remingtonc reviewed (approved):
Approved, with a suggestion.

BenderScript committed:
Parameterized makefile

- Preparing makefile for new project layout
- Included parameters for mostly everything so it is easy to change to new project layout
- Dockerfile creation/removal is under Makefile now
- No need for tools/test/run.sh anymore
